test/acceptance/setup-terraform/ec2.tf

```diff
@@ -0,0 +1,64 @@
+// Setup EC2 container instances for EC2 launch type tests
+data "aws_ssm_parameter" "ecs_optimized_ami" {
+  name = "/aws/service/ecs/optimized-ami/amazon-linux-2/recommended/image_id"
+}
+
+data "aws_security_group" "vpc_default" {
+  name   = "default"
+  vpc_id = module.vpc.vpc_id
+}
+
+locals {
+  esc_optimized_ami = nonsensitive(data.aws_ssm_parameter.ecs_optimized_ami.value)
+}
+
+resource "aws_iam_role" "instance_role" {
+  name = local.name
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Principal = {
+          Service = "ec2.amazonaws.com"
+        }
+      },
+    ]
+  })
+
+  managed_policy_arns = [
+    "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role"
+  ]
+}
+
+resource "aws_iam_instance_profile" "instance_profile" {
+  name = local.name
+  role = aws_iam_role.instance_role.name
+}
+
+resource "aws_instance" "instances" {
+  count = var.instance_count
+
+  ami                  = local.esc_optimized_ami
+  instance_type        = var.instance_type
+  iam_instance_profile = aws_iam_instance_profile.instance_profile.name
+  // Spread instances across subnets
+  subnet_id              = module.vpc.private_subnets[count.index % length(module.vpc.private_subnets)]
+  vpc_security_group_ids = [data.aws_security_group.vpc_default.id]
+
+  user_data = <<EOF
+#!/bin/bash
+echo ECS_CLUSTER=${aws_ecs_cluster.ec2.name} >> /etc/ecs/ecs.config
+EOF
+
+  tags = merge(var.tags, {
+    Name = local.name
+  })
+}
+
+resource "aws_ecs_cluster" "ec2" {
+  name = "${local.name}-ec2"
+  tags = var.tags
+}
```

Comment:
In order to run Fargate and EC2 tests in parallel, I create a separate cluster for the EC2 tests. 

Technically, a single cluster with both Fargate and EC2 capacity could work. But ECS clusters are free and it felt safer with separate clusters.

Reply:
Oh, and `setup-terraform` will always setup both EC2 instances as-is. Do we want a variable so we can choose between Fargate or EC2 or both?

Reply:
Initially, I was imagining that we would have two CI jobs, and each of them would run either only Fargate or EC2. That way we can have one blueprint that we can run in multiple configurations. 

I think the main question here to help decide is if we will have other tests that we'd want to test on both of these launch types (e.g any of the gateways). In that case, I think it'd be easier to configure a CI job and pass some flag to tests to run them in either configuration rather than adding the test for each launch type for each feature. WDYT?

Reply:
Yeah, I went for test parallelization at the golang level so that we can run tests locally in parallel, too. I can't run tests locally with just a CircleCI yaml (or actually, [maybe we can?](https://circleci.com/docs/2.0/local-cli/))

In any case, we can split these into separate acceptance test jobs in CircleCI:
* The CircleCI jobs would run `setup-terraform` and pass in the launch type (which would be a new variable).
* The CircleCI jobs would select EC2/Fargate with test filters: `go test -run 'TestEC2.*' ...`.

>  In that case, I think it'd be easier to configure a CI job and pass some flag to tests to run them in either configuration rather than adding the test for each launch type for each feature.

🤔  I'm tired right now, so maybe not thinking this through - but adding golang test functions which I can run locally seems easier than needing to configure/test a new CI job for each new feature? We can definitely discuss post-Beta release!

Reply:
@ishustava Okay, I've thought about this some more and I agree! It should be simpler, and I don't think I'd actually want to run tests for all launch types locally all the time anyway.